tests/unit/cn/hblock_reader_test.c

```diff
@@ -21,459 +21,290 @@
+    MOCK_SET_FN(mblk_desc, mblk_madvise, mock_mblk_madvise);
+    MOCK_SET_FN(mblk_desc, mblk_madvise_pages, mock_mblk_madvise_pages);
 
-    err = mpool_mblock_delete(mpool, blkid);
-    ASSERT_EQ(0, merr_errno(err));
+    return 0;
 }
 
-MTF_DEFINE_UTEST_PREPOST(hblock_reader_test, madvise_wbt_leaf_nodes_success, test_pre, test_post)
+int
+test_post(struct mtf_test_info *info)
 {
-    merr_t err;
-    struct hdr hdr;
-    struct mpool *mpool = (void *)-1;
-    struct kvs_mblk_desc blk_desc;
-    struct wbt_desc wbt_desc;
-    uint64_t blkid;
-
-    err = mpm_mblock_alloc(FAKE_BLOCK_SIZE, &blkid);
-    ASSERT_EQ(0, merr_errno(err));
-    blk_desc.mbid = blkid;
-
-    err = mpm_mblock_write(blkid, fake_mblock_buf, 0, FAKE_BLOCK_SIZE);
-    ASSERT_EQ(0, merr_errno(err));
-
-    err = mpool_mcache_mmap(mpool, 1, &blk_desc.mbid, &blk_desc.map);
-    ASSERT_EQ(0, merr_errno(err));
-
-    init_hdr(&hdr);
-    err = mpm_mblock_write(blk_desc.mbid, hdr.data, 0, HBLOCK_HDR_PAGES * PAGE_SIZE);
-    ASSERT_EQ(0, merr_errno(err));
-
-    hbr_madvise_wbt_leaf_nodes(&blk_desc, &wbt_desc, MADV_WILLNEED);
+    return 0;
+}
 
-    wbt_desc.wbd_n_pages = 0;
-    wbt_desc.wbd_leaf_cnt = 0;
-    hbr_madvise_wbt_leaf_nodes(&blk_desc, &wbt_desc, MADV_WILLNEED);
+MTF_BEGIN_UTEST_COLLECTION(hblock_reader_test)
 
-    mpool_mcache_munmap(blk_desc.map);
+MTF_DEFINE_UTEST_PREPOST(hblock_reader_test, t_hbr_read_metrics, test_pre, test_post)
+{
+    merr_t err;
+    struct hblk_metrics s;
 
-    err = mpool_mblock_delete(mpool, blkid);
-    ASSERT_EQ(0, merr_errno(err));
+    err = hbr_read_metrics(&mblk, &s);
+    ASSERT_EQ(err, 0);
 }
 
-MTF_DEFINE_UTEST_PREPOST(hblock_reader_test, read_desc_error, test_pre, test_post)
+MTF_DEFINE_UTEST_PREPOST(hblock_reader_test, t_hbr_read_ptree_region_desc, test_pre, test_post)
 {
     merr_t err;
-    struct mpool *mpool = (void *)-1;
-    struct mpool_mcache_map *map = (void *)-1;
-    const uint64_t blkid = 0xffff;
-    struct kvs_mblk_desc desc;
-    struct mblock_props props;
+    struct wbt_desc s;
 
-    mapi_inject_ptr(mapi_idx_mpool_mcache_getbase, NULL);
+    err = hbr_read_ptree_region_desc(&mblk, &s);
+    ASSERT_EQ(err, 0);
 
-    err = hbr_read_desc(mpool, map, &props, blkid, &desc);
-    ASSERT_NE(0, merr_errno(err));
+    ASSERT_EQ(s.wbd_first_page, hbh_ro.hbh_ptree_data_off_pg);
+    ASSERT_EQ(s.wbd_n_pages,    hbh_ro.hbh_ptree_data_len_pg);
 
-    mapi_inject_unset(mapi_idx_mpool_mcache_getbase);
+    ASSERT_EQ(s.wbd_version,    hbh_ro.hbh_ptree_hdr.wbt_version);
+    ASSERT_EQ(s.wbd_root,       hbh_ro.hbh_ptree_hdr.wbt_root);
+    ASSERT_EQ(s.wbd_leaf,       hbh_ro.hbh_ptree_hdr.wbt_leaf);
+    ASSERT_EQ(s.wbd_leaf_cnt,   hbh_ro.hbh_ptree_hdr.wbt_leaf_cnt);
+    ASSERT_EQ(s.wbd_kmd_pgc,    hbh_ro.hbh_ptree_hdr.wbt_kmd_pgc);
 }
 
-MTF_DEFINE_UTEST_PREPOST(hblock_reader_test, read_desc_success, test_pre, test_post)
+MTF_DEFINE_UTEST_PREPOST(hblock_reader_test, t_hbr_read_seqno_range, test_pre, test_post)
```

Comment:
Don't really like the t_ prefix

Reply:
we've had this discussion.  that field becomes a function name.  prefixing it helps avoid name conflicts, and is nice for debugging to identify test entry points.